spec/datadog/tracing/contrib/rails/cache_spec.rb

```diff
@@ -117,6 +117,34 @@
       expect(set.get_tag(Datadog::Tracing::Metadata::Ext::TAG_OPERATION))
         .to eq('cache')
     end
+
+    context 'when cache_key_enabled is false' do
+      before do
+        Datadog.configuration.tracing[:active_support][:cache_key_enabled] = false
+      end
+
+      expect(read).to eq(50)
+
+      expect(spans).to have(2).items
+      get, set = spans
+      expect(get.name).to eq('rails.cache')
+      expect(get.type).to eq('cache')
+      expect(get.resource).to eq('GET')
+      expect(get.service).to eq('rails-cache')
+      expect(get.get_tag('rails.cache.backend')).to eq('file_store')
+      expect(get.get_tag('rails.cache.key')).to be_nil
+      expect(set.name).to eq('rails.cache')
+
+      expect(get.get_tag(Datadog::Tracing::Metadata::Ext::TAG_COMPONENT))
+        .to eq('active_support')
+      expect(get.get_tag(Datadog::Tracing::Metadata::Ext::TAG_OPERATION))
+        .to eq('cache')
+
+      expect(set.get_tag(Datadog::Tracing::Metadata::Ext::TAG_COMPONENT))
+        .to eq('active_support')
+      expect(set.get_tag(Datadog::Tracing::Metadata::Ext::TAG_OPERATION))
+        .to eq('cache')
+    end
```

Comment:
I recommend only testing for the relevant behavior that this test wants to cover, so that we understand can easily debug failures in it, without being concerned with fields that are already test elsewhere:
```suggestion
    context 'when cache_key_enabled is false' do
      before do
        Datadog.configuration.tracing[:active_support][:cache_key_enabled] = false
      end

      expect(read).to eq(50)

      expect(spans).to have(2).items
      get, set = spans
      expect(get.name).to eq('rails.cache')
      expect(get.get_tag('rails.cache.key')).to be_nil
    end
```